airflow/www/views.py

```diff
@@ -3796,11 +3796,11 @@ def historical_metrics_data(self):
             data = {
                 "dag_run_types": {
                     **{dag_run_type.value: 0 for dag_run_type in DagRunType},
-                    **{run_type: sum_value for run_type, sum_value in dag_run_types},
+                    **dag_run_types,
                 },
                 "dag_run_states": {
                     **{dag_run_state.value: 0 for dag_run_state in DagRunState},
-                    **{run_state: sum_value for run_state, sum_value in dag_run_states},
+                    **dict(dag_run_states),
```

Comment:
```suggestion
                    **dag_run_states,
```

Reply:
dag_run_states and dag_run_types are sequences of tuples (tuple of size 2), so we need to call dict()